### PR TITLE
Multiple fixes

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
* Allow loading folders
 * Add explicit brackets after types links
 * Add --no-config option and use .metadocrc as default config
 * Do not generate extra lines between functions
 * Make title optional

/cc @lundibundi @belochub @nechaido

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -50,21 +50,53 @@ const args = yargs
     type: 'string',
     describe: 'custom config file',
   })
+  .option('no-config', {
+    alias: 'n',
```
Comment (lundibundi) [inline bin/cli.js]: I wouldn't add an alias for this, I think this is rare enough.

File: bin/cli.js [hunk 2]
```diff
@@ -50,21 +50,53 @@ const args = yargs
...
+    type: 'boolean',
+    describe: 'custom config file',
```
Comment (lundibundi) [inline bin/cli.js]:
```suggestion
    describe: 'do not look for a configuration file',
```

File: bin/cli.js [hunk 3]
```diff
@@ -50,21 +50,53 @@ const args = yargs
...
+};
+
 const getConfig = args => {
   let cfg = {};
-  if (args.config && fs.existsSync(args.config)) {
-    cfg = JSON.parse(fs.readFileSync(args.config));
-  } else if (args.config === '') {
-    cfg = JSON.parse(fs.readFileSync('.metadocrc'));
+  if (!args.noConfig) {
+    if (args.config && fs.existsSync(args.config)) {
```
Comment (lundibundi) [inline bin/cli.js]: Perhaps `args.config = args.config || '.metadocrc'`.

File: bin/cli.js [hunk 4]
```diff
@@ -50,21 +50,53 @@ const args = yargs
...
+      cfg = JSON.parse(fs.readFileSync(args.config));
+    } else if (fs.existsSync('.metadocrc')) {
```
Comment (lundibundi) [inline bin/cli.js]: It shouldn't look for the default config file if args.config is provided.

File: lib/introspector.js [hunk 5]
```diff
@@ -204,9 +204,17 @@ const parseFunction = (fn, lines) => {
     comments: [],
   };
 
-  if (lines[0] && !lines[0].startsWith('Signature:')) sig.title = lines[0];
+  if (
```
Comment (tshemsedinov) [inline lib/introspector.js]:
```suggestion
  const first = lines[0];
  if (
```
Condition will be:
```js
  if (first &&
    first.startsWith('Signature:') &&
    !isNamedLine(first) &&
    !first.startsWith(' ')
  )
```

File: lib/introspector.js [hunk 6]
```diff
@@ -204,9 +204,17 @@ const parseFunction = (fn, lines) => {
...
+    lines[0] &&
+    !lines[0].startsWith('Signature:') &&
+    !isNamedLine(lines[0]) &&
+    !lines[0].startsWith(' ')
+  ) {
+    sig.title = lines[0];
```
Comment (tshemsedinov) [inline lib/introspector.js]:
```suggestion
    sig.title = first;
```

File: package.json [hunk 7]
```diff
@@ -21,9 +21,9 @@
   },
   "readmeFilename": "README.md",
   "scripts": {
-    "doc": "./bin/cli.js -c",
+    "doc": "./bin/cli.js",
     "lint": "eslint .",
-    "test": "eslint . && ./bin/cli.js -d test/ test/example.js"
+    "test": "eslint . && ./bin/cli.js --no-config=true -d test/ test/example.js"
```
Comment (belochub) [inline package.json]: Can't we use it like that?
```suggestion
    "test": "eslint . && ./bin/cli.js --no-config -d test/ test/example.js"
```

File: bin/cli.js [hunk 8]
```diff
@@ -50,21 +50,51 @@ const args = yargs
...
+  return result;
+};
+
 const getConfig = args => {
   let cfg = {};
-  if (args.config && fs.existsSync(args.config)) {
-    cfg = JSON.parse(fs.readFileSync(args.config));
-  } else if (args.config === '') {
-    cfg = JSON.parse(fs.readFileSync('.metadocrc'));
+  if (args.config) {
```
Comment (lundibundi) [inline bin/cli.js]: Will this use '.metadocrc' for `metadoc test`?

Comment (SemenchenkoVitaliy) [reply bin/cli.js]: `metadoc` will use `.metadocrc` unless you specify `--no-config` flag or another config file with `--config`.

Comment (lundibundi) [reply bin/cli.js]: I've just run `./bin/cli.js` on this branch and got `No files were specified. Exiting...` even though we have `.metadocrc` in the package root folder.

Comment (SemenchenkoVitaliy) [reply bin/cli.js]: Fixed in https://github.com/metarhia/metadoc/pull/22/commits/c457b2fbd71465f209412a91d5833e66597aeac9.

File: bin/cli.js [hunk 9]
```diff
@@ -50,21 +50,51 @@ const args = yargs
...
+  return result;
+};
+
 const getConfig = args => {
   let cfg = {};
-  if (args.config && fs.existsSync(args.config)) {
-    cfg = JSON.parse(fs.readFileSync(args.config));
-  } else if (args.config === '') {
-    cfg = JSON.parse(fs.readFileSync('.metadocrc'));
+  if (typeof args.config === 'string' || args.config === undefined) {
```
Comment (lundibundi) [inline bin/cli.js]:
```suggestion
  if (args.config !== false) {
```

File: doc/doc.md [hunk 10]
```diff
@@ -82,11 +80,10 @@ See GitHub for a full [list of contributors](https://github.com/metarhia/metadoc
 
 Licesed under MIT license. Copyright (c) 2018 Metarhia contributors
 
-
-[`<Object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
-[`<Function>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function
-[`<RegExp>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
-[`<Map>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
-[`<Array>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
+[`<object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
```
Comment (lundibundi) [inline doc/doc.md]: These md links look outdated.

**Comment by lundibundi** [comment]
Landed in https://github.com/metarhia/metadoc/commit/af7b0b9bdef4a2b58aeefcac9de0b394e8225362.

**Pull request closed without merging**